tmol/score/omega/potentials/omega_pose_score.hh

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include <tmol/utility/tensor/TensorAccessor.h>
+#include <tmol/utility/tensor/TensorPack.h>
+#include <tmol/utility/tensor/TensorStruct.h>
+#include <tmol/utility/tensor/TensorUtil.h>
+#include <tmol/utility/nvtx.hh>
+
+#include <tmol/score/common/accumulate.hh>
+#include <tmol/score/common/geom.hh>
+#include <tmol/score/common/tuple.hh>
+
+#include "omega.hh"
+#include "params.hh"
+// #include "rotamer_pair_energy_lj.hh"
+
+namespace tmol {
+namespace score {
+namespace omega {
+namespace potentials {
+
+template <typename Real, int N>
+using Vec = Eigen::Matrix<Real, N, 1>;
+
+template <
+    template <tmol::Device>
+    class DeviceOps,
+    tmol::Device D,
+    typename Real,
+    typename Int>
+struct OmegaPoseScoreDispatch {
+  static auto f(
+      TView<Vec<Real, 3>, 2, D> coords,
+      TView<Int, 2, D> pose_stack_block_coord_offset,
+      TView<Int, 2, D> pose_stack_block_type,
+      TView<Vec<Int, 2>, 3, D> pose_stack_inter_block_connections,
+      TView<Int, 3, D> block_type_omega_quad_uaids,
+      TView<Int, 3, D> block_type_atom_downstream_of_conn,
+
+      TView<OmegaGlobalParams<Real>, 1, D> global_params,
+
+      bool compute_derivs
+
+      ) -> std::tuple<TPack<Real, 1, D>, TPack<Vec<Real, 3>, 2, D>>;
```

Comment:
The convention for pose scores is that each term should return a 2D tensor of values:
1st dimension: the number of sub-terms defined (e.g. lk-ball defines 4, fa_elec defines 1)
2nd dimension: the number of poses in the stack.
That way the set of 2D tensors can be concatenated along their first dimensions and then the weighted scores  computed by a matrix-vector dot product (see tmol/score/score_function.py:186)

Reply:
Ok, I was wondering about that, as the shape of the baseline I copied from LJLK didn't match. Thanks!